### PR TITLE
cdc,retry: fix leader missing by extending region retry duration

### PR DESCRIPTION
**Pull request by overvenus**
### What problem does this PR solve?

Issue Number: close #5230

### What is changed and how it works?

* Fix replication interruption due to leader missing by extending region retry duration

Other changes
* Remove `oldvalue` in puller and kv client.
* Refine retry error info, includes the original error.

### Check List 

Tests 

 - Unit test

Related changes

 - Need to cherry-pick to the release branch

### Release note 

```release-note
Fix replication interruption due to leader missing by extending region retry duration
```

### CODE REVIEW

File: cdc/kv/client.go [hunk 1]
```diff
@@ -961,7 +959,8 @@ func (s *eventFeedSession) divideAndSendEventFeedToRegions(
 				zap.Reflect("regions", metas),
 				zap.String("changefeed", s.client.changefeed))
 			return nil
-		}, retry.WithBackoffMaxDelay(50), retry.WithMaxTries(100), retry.WithIsRetryableErr(cerror.IsRetryableError))
```
Comment (3AceShowHand) [inline cdc/kv/client.go:964]: it looks that this is the root cause of the problem.

how about add some comment about retry logic here.

**Review by 3AceShowHand — APPROVED** [review]

File: cdc/kv/client.go [hunk 2]
```diff
@@ -294,7 +294,6 @@ type CDCKVClient interface {
 		ctx context.Context,
 		span regionspan.ComparableSpan,
 		ts uint64,
-		enableOldValue bool,
```
Comment (nongfushanquan) [inline cdc/kv/client.go:297]: why remove the enableOldValue flag in this PR?  Is there  an issue related to this item?

File: pkg/retry/retry_with_opt.go [hunk 3]
```diff
@@ -47,6 +47,14 @@ func run(ctx context.Context, op Operation, retryOption *retryOptions) error {
 	default:
 	}
 
+	if retryOption.totalRetryDuration != time.Duration(0) {
+		maxTries := getApproximateMaxTryBasedOnTotalRetryDuration(
```
Comment (CharlesCheung96) [inline pkg/retry/retry_with_opt.go:51]: Why not just manage the total retry duration here?

File: cdc/kv/client.go (hunk 1 above)
Comment (overvenus) [reply cdc/kv/client.go:964]: Comments added in kv client config.

File: cdc/kv/client.go (hunk 2 above)
Comment (overvenus) [reply cdc/kv/client.go:297]: The `enableOldValue ` param has been deprecated monthes ago, it's a cleanup for dead code.

File: pkg/retry/options.go [hunk 4]
```diff
@@ -66,18 +70,19 @@ func WithBackoffMaxDelay(delayInMs int64) Option {
 }
 
 // WithMaxTries configures maximum tries, if tries <= 0 "defaultMaxTries" will be used
```
Comment (asddongmen) [inline pkg/retry/options.go:72]: `if tries <= 0 "defaultMaxTries" will be used` seems wrong.

File: pkg/retry/options.go [hunk 5]
```diff
@@ -15,14 +15,16 @@ package retry
...
 )
 
 const (
 	// defaultBackoffBaseInMs is the initial duration, in Millisecond
 	defaultBackoffBaseInMs = 10.0
 	// defaultBackoffCapInMs is the max amount of duration, in Millisecond
-	defaultBackoffCapInMs = 100.0
-	defaultMaxTries       = 3
+	defaultBackoffCapInMs   = 100.0
+	defaultMaxTries         = math.MaxInt64
```
Comment (asddongmen) [inline pkg/retry/options.go:26]: Why do you set this to `math.MaxInt64`?

**Review by asddongmen — APPROVED** [review]

**Review by CharlesCheung96 — APPROVED** [review]

**Pull request merged**